lua: provide whether ASAN build in tarantool.build.asan (#9062)
tarantool/tarantool: changelogs/unreleased/add-asan-flag-to-tarantool-build.md

## feature/box

> * A new flag `tarantool.build.asan` is added showing whether build
>   flag `ENABLE_ASAN` is set or not.

Reviewer: ```suggestion
* Added a  new flag `tarantool.build.asan` that shows whether build
  flag `ENABLE_ASAN` is set.
```

Author: Fixed.
